thunder/benchmarks/__init__.py

```diff
@@ -2940,6 +2941,58 @@ def fn(self) -> Callable:
         return model
 
 
+class TorchbenchBenchmark(Benchmark, metaclass=UserFacingBenchmarkMeta):
+    _args = (
+        BenchmarkArg(
+            name="module_name",
+            description="The torchbenchmark module name (str).",
+        ),
+        BenchmarkArg(
+            name="device",
+            description="A device (str) to run on {'cpu' | 'cuda'}. Default is 'cuda'.",
+        ),
+        BenchmarkArg(
+            name="requires_grad",
+            description="Whether the model parameters require grad. Default is True.",
+        ),
+    )
+
+    @classmethod
+    @property
+    def name(cls) -> str:
+        return "torchbench"
+
+    @classmethod
+    @property
+    def description(cls) -> str:
+        return "Torchbench fixture"
+
+    @classmethod
+    @property
+    def args(cls) -> tuple[BenchmarkArg, ...]:
+        return cls._args
+
+    def __init__(self, module_name, device: str = "cuda", requires_grad: bool = True):
+        import importlib
+
+        module = importlib.import_module(f"torchbenchmark.models.{module_name}")
+        self.benchmark_cls = getattr(module, "Model", None)
+
+        benchmark = self.benchmark_cls(test="train" if requires_grad else "eval", device=device)
+
+        model, example = benchmark.get_module()
+        self.model = model
+        self.example_input = example
+
+    def make_batch(self) -> tuple[list, dict]:
```

Comment:
Unrelated and non-actionable for this PR but I wonder why `make_batch` as per type-signature from parent class `Benchmark` returns `tuple[list, dict]` instead of `tuple[tuple, dict]` as `args` are usually represented with tuple.

Also, other Benchmark classes return `tuple[tuple, dict]`

https://github.com/Lightning-AI/lightning-thunder/blob/968d6d07bf58827a191dfe3e01ce04e6e65a3a09/thunder/benchmarks/__init__.py#L1035-L1037

https://github.com/Lightning-AI/lightning-thunder/blob/968d6d07bf58827a191dfe3e01ce04e6e65a3a09/thunder/benchmarks/__init__.py#L1234

https://github.com/Lightning-AI/lightning-thunder/blob/968d6d07bf58827a191dfe3e01ce04e6e65a3a09/thunder/benchmarks/__init__.py#L1406

Reply:
That is true, i found that creating the list was the easiest to express arguments as sometimes they are lists from torchbench, however it's a good idea and I will keep it in mind for the next PR